cdap-app-fabric/src/main/java/io/cdap/cdap/internal/app/services/ProgramNotificationSubscriberService.java

```diff
@@ -545,7 +546,10 @@ private void processWorkflowOnStop(AppMetadataStore appMetadataStore,
                                      Notification notification,
                                      byte[] sourceId, List<Runnable> runnables) throws Exception {
     ApplicationId appId = programRunId.getParent().getParent();
-    WorkflowSpecification workflowSpec = Optional.ofNullable(appMetadataStore.getApplication(appId))
+    ApplicationMeta appMetadata = ApplicationId.DEFAULT_VERSION.equals(appId.getVersion())
```

Comment:
we shouldn't need to be doing this here, the app version is supposed to be accurate. Is that not the case?